responsibleai/tests/causal/test_causal_manager.py

```diff
@@ -27,6 +27,30 @@ def test_causal_no_categoricals(self, housing_data):
         assert len(result.config.treatment_features) == 1
         assert result.config.treatment_features[0] == 'AveRooms'
 
+    def test_causal_manager_cohort_effects(self, housing_data):
+        train_df, test_df, target_feature = housing_data
+
+        manager = CausalManager(train_df, test_df, target_feature,
+                                ModelTask.REGRESSION, None)
+        manager.add(['AveRooms'])
+        manager.compute()
+
+        id = manager.get()[0].id
+        X_test = test_df.drop(target_feature, axis=1)
+        global_effect_X_test = manager._cohort_effects(id, X_test)
+
+        assert len(global_effect_X_test) == 1
+        EFFECTS_ATTRIBUTES = [
+            'point',
+            'stderr',
+            'zstat',
+            'ci_lower',
+            'ci_upper',
+            'p_value',
+        ]
+        for effect in global_effect_X_test:
+            assert all(attr in effect for attr in EFFECTS_ATTRIBUTES)
```

Comment:
I wonder, is there some way to assert that the statistics are indeed for that specific cohort and not for the overall all data group?  Maybe something as simple as looking at the effects and validating that the values are different from the all data cohort, or for a better test, perhaps even validating the values shifted based on the instances we reduced to in the dataset (maybe then you have to choose the rows with a filter instead of random sample in order to make it easier to get consistent results and test for this).  Not necessary for this PR, but I think it would be good to have some tests for cohorts like that.  I suppose our other components should have similar tests if we don't have them already, to validate that our computed statistics do indeed change as we expect them to for specific cohorts.

Reply:
Now that we have cohort filtering is responsibleai, we should have some tests around cohort filtering. But as you said, this is outside of scope of this PR.